src/cudamatrix/cu-device.h

```diff
@@ -26,6 +26,7 @@
 #if HAVE_CUDA == 1
 
 #include <cublas_v2.h>
+#include <cudnn.h>
```

Comment:
You should wrap this, and I think all of the code that uses the cudnn context in cu-device.cc in this:
```
#ifdef HAVE_CUDNN
code goes here...
#endif
```

You could define the HAVE_CUDNN macro somewhere in src/configure by adding a line like "CXX_FLAGS += -DHAVE_CUDNN" to kaldi.mk

The point is that I don't think we can reasonably assume most users of Kaldi will want or need cudnn. So we have to have these ugly macros to support compiling both with and without cudnn.